app/redux/wallet/actions.ts

```diff
@@ -103,22 +104,32 @@ export const readWalletFiles = () => async (dispatch: AppThDispatch) => {
 };
 
 export const createNewWallet = ({
-  existingMnemonic = '',
+  mnemonic,
   password,
   apiUrl,
   genesisID,
   type,
   name,
 }: {
-  existingMnemonic?: string | undefined;
   password: string;
   type: WalletType;
   apiUrl: SocketAddress | null;
   genesisID: string;
   name?: string;
+  mnemonic: {
+    existing?: string;
+    generate?: MnemonicStrengthType;
+  };
```

Comment:
Why you used the type not so transparent as I described before? :(

Can you do something like this?
```
type MnemonicExisting = { existing: string };
type MnemonicNew = { strength: 12 | 24 };
type MnemonicOpts = MnemonicExisting | MnemonicNew;

const isMnemonicExisting = (mnemonic: any): mnemonic is MnemonicExisting =>
  mnemonic && typeof mnemonic.existing === 'string';

const isMnemonicNew = (mnemonic: any): mnemonic is MnemonicNew =>
  mnemonic && (mnemonic.strength === 12 || mnemonic.strength === 24);

const create = (mnemonic: MnemonicOpts) => {
  if (isMnemonicExisting(mnemonic)) {
    console.log('existing:', mnemonic.existing);
  } else if (isMnemonicNew(mnemonic)) {
    console.log('new using strength', mnemonic.strength);
  } else {
    console.log('fallback to default', 12);
  }
};

create({ existing: 'lalala' });
create({ strength: 12 });
create({ strength: 24 });

const t = JSON.parse('{}');
create(t);
```

Reply:
And please put it in a separate module (shared) and reuse then in Renderer and Main processes.